utils/cloud_metadata.py

```diff
@@ -226,3 +229,34 @@ def get_instance_id(agentConfig):
             return EC2.get_metadata(agentConfig).get("instance-id", None)
         except Exception:
             return None
+
+class CloudFoundry(object):
+    host_aliases = []
+
+    @staticmethod
+    def get_host_aliases(agentConfig):
+        if not CloudFoundry.is_cloud_foundry(agentConfig):
+            return CloudFoundry.host_aliases
+        if os.environ.get("DD_BOSH_ID"):
+            CloudFoundry.host_aliases.append(os.environ.get("DD_BOSH_ID"))
+        if agentConfig.get("bosh_id"):
+            CloudFoundry.host_aliases.append(agentConfig.get("bosh_id"))
+        if len(CloudFoundry.host_aliases) == 0:
+            # Only use this if the prior one fails
+            # The reliability of the socket hostname is not assured
+            CloudFoundry.host_aliases.append(socket.gethostname())
+        return []
+
+    @staticmethod
+    def is_cloud_foundry(agentConfig):
+        if agentConfig.get("cloud_foundry"):
+            return True
```

Comment:
ok, thanks for the explanation.

So the one thing I want to make sure of here is that we're not too "optimistic" in our detection of a cloud foundry environment, i.e. we don't detect that we're running on cloud foundry when we're not. Even though for now this is only used to add a host alias (including the socket hostname if no bosh-related alias is defined), given the name of the function I could see this being used in the future for other things :)

Or, we can keep the detection "optimistic" and add a comment that makes that explicit